### ISSUE TITLE
My lock screen is black when I wanna unlock my phone via pattern.

### ISSUE THREAD

**Issue by Septemberlemon** [Genymobile/scrcpy#5625]
This is my phone's screen lock:
![image](https://github.com/user-attachments/assets/594889e7-53d8-4f73-9d0c-b11b70461578)
and then I drag the screen up,on the real phone it goes like this:
![WIN_20241206_22_32_33_Pro](https://github.com/user-attachments/assets/49b60a70-cb58-490a-90d5-dae1f0586db6)
on the scrcpy it goes black like this:
![image](https://github.com/user-attachments/assets/5d810a49-bb5c-4cbc-b9fe-8ade33904bb8)
and I can't draw the pattern or unlock it via fingerprint
Now I had to unlock my phone first,and then connect to it,it's a little bit inconvenient,even if it's not a big trouble
Can someone tell me how to solve that?    (^_^)

**Comment by rom1v** [comment]
The lock screen is "protected" by secure flag. It cannot be mirrored anymore with shell permissions since Android 12:  #36 #2129.

**Comment by Septemberlemon** [comment]
> The lock screen is "protected" by secure flag. It cannot be mirrored anymore with shell permissions since Android 12: #36 #2129.

so it's impossible to do that?Q^Q

**Comment by rom1v** [comment]
Yes, Android prevents to capture the screen when the "secure flag" is enabled.

**Comment by VoidNakamura** [comment]
> Yes, Android prevents to capture the screen when the "secure flag" is enabled.

Is it possible to bypass this without root? Would a USB-to-HDMI cable work?

**Comment by parkerlreed** [comment]
> > Yes, Android prevents to capture the screen when the "secure flag" is enabled.
> 
> Is it possible to bypass this without root? Would a USB-to-HDMI cable work?

Yes and always has. One issue is if you have "Automatically start Dex when HDMI connected" your lock screen would still be on the phone itself, so that option would have to be off to get the phone screen itself to show up.

**Comment by VincentChalnot** [comment]
I'm surprised no one mentioned the possibility to clear the pattern lock with adb:
```shell
$ adb locksettings clear --old XXXXXX
```
Where XXXX is corresponds to the list of "cells" of your pattern.

**Comment by parkerlreed** [comment]
List as in their number? 1-9?

**Comment by rom1v** [comment]
https://nerdschalk.com/change-remove-lockscreen-pattern-pin-password-adb-android-oreo/

**Comment by ConcurrentCrab** [comment]
> I'm surprised no one mentioned the possibility to clear the pattern lock with adb:
> 
> $ adb locksettings clear --old XXXXXX
> Where XXXX is corresponds to the list of "cells" of your pattern.

@VincentChalnot thank you for your comment :D you saved someone's precious memories today. (and I got a lesson to make timely backups)

Tried about everything else on these forums, but none of the `input keyevent` commands were working, or maybe I just couldn't see any feedback.

One note is that the command may error out if you have a work profile set up which disallows a no-password lockscreen (There's a setting to have a separate work profile password from the lockscreen password, not sure if that would've helped in this situation).

I had to:
```
~ $ adb shell pm list users
Users:
        UserInfo{0:Owner:ABCD} running
        UserInfo{10:Work profile:WXYZ} running
~ $ adb shell pm remove-user 10
Success: removed user
```

Lost my MFA stuff in the work profile, but oh well, there are people getting paid to help me set those up again.

**Comment by kkuhle** [comment]
If you have root/lsposed, you can use https://github.com/Xposed-Modules-Repo/io.github.lsposed.disableflagsecure (works for me on tested situations such as lockscreen and bitwarden)
